cunumeric/module.py

```diff
@@ -523,6 +523,112 @@ def tile(a, reps):
     return result
 
 
+# Spliting arrays
+@copy_docstring(np.vsplit)
+def vsplit(a, indices):
+    return split(a, indices, axis=0)
+
+
+@copy_docstring(np.hsplit)
+def hsplit(a, indices):
+    return split(a, indices, axis=1)
+
+
+@copy_docstring(np.dsplit)
+def dsplit(a, indices):
+    return split(a, indices, axis=2)
+
+
+@copy_docstring(np.split)
+def split(a, indices, axis=0):
+    return array_split(a, indices, axis, equal=True)
+
+
+@copy_docstring(np.array_split)
+def array_split(a, indices, axis=0, equal=False):
+    array = ndarray.convert_to_cunumeric_ndarray(a)
+    dtype = type(indices)
+    split_pts = []
+
+    if dtype == int:
+        res = array.shape[axis] % indices
+        len_subarr = array.shape[axis] // indices
+        first_idx = len_subarr
+        end_idx = array.shape[axis]
+
+        if array.ndim >= axis:
```

Comment:
This should be `array.ndim > axis`

Reply:
axis starts from 0 so when the array is 3D, the maximum valid value for axis is 2.

Reply:
Correct. For a 3d array `array.ndim == 3`, so the non-error path should only be taken for `axis < 3`.

Reply:
Actually if `axis >= array.ndim`, this code already crashes at the `array.shape[axis]` access on line 554. This check should be moved to the beginning of the function, to also cover the cases where `indices` is not an integer.

Reply:
Got it. I'll update the code as you suggested.